Common/src/linear_algebra/CSysSolve.cpp

```diff
@@ -972,7 +959,30 @@ unsigned long CSysSolve<ScalarType>::Solve(CSysMatrix<ScalarType> & Jacobian, co
         break;
     }
 
+    /*--- Start recording if it was stopped for the linear solver ---*/
+#ifdef CODI_REVERSE_TYPE
+    AD::StartRecording();
+    SU2_OMP_BARRIER
+
+    SU2_OMP_MASTER {
+      AD::SetExtFuncOut(&LinSysSol[0], LinSysSol.GetLocSize());
+      AD::FuncHelper->addUserData(&LinSysRes);
+      AD::FuncHelper->addUserData(&LinSysSol);
+      AD::FuncHelper->addUserData(&Jacobian);
+      AD::FuncHelper->addUserData(geometry);
+      AD::FuncHelper->addUserData(config);
+      AD::FuncHelper->addUserData(this);
+    }
+    END_SU2_OMP_MASTER
+    SU2_OMP_BARRIER
+
+    AD::FuncHelper->addToTape(CSysSolve_b<ScalarType>::Solve_b);
+    SU2_OMP_BARRIER
```

Comment:
Registering inputs, outputs and user data on one `codi::ExternalFunctionHelper` is not thread-safe. However, all participating threads add the external function to their tape. Likewise, all participating threads make calls to `AD::StartRecording` and `AD::StopRecording`.